tutorials/layout/index.md

```diff
@@ -0,0 +1,1003 @@
+everything is a widget&mdash;even layout models are widgets.
+The images, icons, and text that you see in a Flutter app  are all widgets.
+But things you don't see are also widgets, such as the rows, columns,
+and grids that arrange, constrain, and align the visible widgets.
+
+You create a layout by composing widgets to build more complex widgets.
+For example, the screenshot on the left shows 3 icons with a label under
+each one:
+
+<img src="images/lakes-icons.png" style="border:1px solid black" alt="sample layout">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="images/lakes-icons-visual.png" style="border:1px solid black" alt="node tree representing sample layout">
+
+{% comment %}
+<ul class="col2">
+<li>
+<img src="images/lakes-icons.png" style="border:1px solid black" alt="">
+</li>
+<li>
+<img src="images/lakes-icons-visual.png" style="border:1px solid black" alt="">
+</li>
+</ul>
+{% endcomment %}
+
+The screenshot on the right displays the visual layout, showing a row of
+3 columns where each column contains an icon and a label.
+
+<aside class="alert alert-info" markdown="1">
+**Note:** Most of the screenshots in this tutorial are displayed with
+`debugPaintSizeEnabled` set to true so you can see the visual layout.
+For more information, see
+[Visual debugging](https://flutter.io/debugging/#visual-debugging), a section in
+[Debugging Flutter Apps](https://flutter.io/debugging/#visual-debugging).
+</aside>
+
+Flutter's widget tree for this UI is as follows:
+
+<img src="images/sample-flutter-layout.png" alt="node tree representing sample layout">
+
+Most of this should look as you might expect, but you might be wondering
+about the Containers (shown in pink). Container is a special widget that allows
+you to customize its child widget. Use a Container when you want to
+add padding, margins, borders, or background color, to name some of its
+capabilities. If you are familiar with web programming, Container is
+similar to an HTML `<div>`.
+
+In this example, each Text widget is placed in a Container to add margins.
+The entire Row is also placed in a Container to align the
+row's children so that the free space in the row is divided evenly between,
+before, and after the children.
+
+The rest of the UI in this example is controlled by properties.
+Set an Icon's color using its `color` property.
+Use Text's `style` property to set the font, its color, weight, and so on.
+Columns and Rows have properties that allow you to specify how their
+children are aligned vertically or horizontally, and how much space
+the children should occupy.
+
+<hr>
+
+<a name="lay-out-a-widget"></a>
+## Lay out a widget
+
+<div class="panel" markdown="1">
+
+#### <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?
+
+* Create an Image, Icon, or Text widget.
+* Add it to a layout widget, such as Center, Align, SizedBox, or ListView,
```

Comment:
The `GridView.extent` version is meant to adapt well to different screen sizes and orientations because it keeps the tiles roughly the same size and just adds more columns if you have more space.